### PR TITLE
Add test for divulgence with keys

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
### Hotfix

Reverts https://github.com/digital-asset/daml/pull/6139 while making sure to not insert divulged contracts deleted in the current transaction.

did it

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by gerolf-da** [comment]
LGTM

Thank you, @stefanobaghino-da!

**Comment by stefanobaghino-da** [comment]
This PR is the result of a joint effort by @fabiotudone-da, @meet-da and @gerolf-da. Many thanks for making this happen.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          actingParties = Some(Set(bob)),
+          signatories = Set(bob),
+          stakeholders = Set(bob),
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          ),
+        ),
+        parent = rootExercise,
+      )
+      val nestedExercise = builder.add(
```
Comment (gerolf-da) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala:88]: Is `nestedExercise` missing the parent assignment?

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          submittingParty = Some(bob),
+          workflowId = None,
+          ledgerEffectiveTime = t2,
+          recordedAt = t2,
+          transaction = tx2,
+          explicitDisclosure = Map.empty,
+        )
+      )
+      _ <- store(
+        divulgedContracts = Map((create1, someVersionedContractInstance) -> Set(alice)),
```
Comment (gerolf-da) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala:164]: The fetch node that is doing the divulgence fetches `create2`. Should this be used here as well?

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala:88]: Addressed by https://github.com/digital-asset/daml/pull/6607/commits/e64699148e955152ef4f31e54b1cb964b1c50fea

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala]: Addressed by https://github.com/digital-asset/daml/pull/6607/commits/070f7dd7949de123f813e57eec30911c7f400f3e

**Pull request merged**